src/azure-cli/azure/cli/command_modules/appconfig/_params.py

```diff
@@ -0,0 +1,148 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (get_enum_type,
+                                                get_location_type, tags_type)
+
+from azure.cli.core.commands.validators import \
+    get_default_location_from_resource_group
+
+from ._validators import (validate_appservice_name_or_id,
+                          validate_connection_string, validate_datetime,
+                          validate_export, validate_import,
+                          validate_import_depth, validate_query_fields,
+                          validate_separator)
+
+
+def load_arguments(self, _):
+    with self.argument_context('appconfig create') as c:
+        c.argument('resource_group_name', options_list=['--resource-group', '-g'], required=True)
+        c.argument('name', options_list=['--name', '-n'], required=True, help='Name of the App Configuration to be created. You can configure the default name using `az configure --defaults app_configuration_store=<name>`', configured_default='app_configuration_store')
```

Comment:
You should not need to provide `required=True`. The CLI can infer this from the command signature.  

Additionally, this should expose `id_part='name'` to enable `--ids`.
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-the-ids-parameter

Reply:
Done.